### PR TITLE
testing/openimageio: new aport

### PR DESCRIPTION
**Pull request by Klowner**
new package💃
(depends on opencolorio)

### CODE REVIEW

File: testing/openimageio/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	cd "$builddir"
+	mkdir build && cd build
+	cmake .. \
+		-DUSE_OPENSSL=ON \
+		-DUSE_PYTHON=ON \
+		-DUSE_PYTHON3=OFF \
+		-DCMAKE_INSTALL_PREFIX=/usr \
+		-DPYLIB_INSTALL_DIR=lib/python2.7/site-packages \
+		-DOIIO_BUILD_TESTS=OFF \
+		-DOIIO_BUILD_TOOLS=ON
```
Comment (jirutka) [inline testing/openimageio/APKBUILD]: Please add `|| return 1` at the end of this line (just this one), to ensure that an error will halt the build (abuild doesn’t run with `set -e` yet).

File: testing/openimageio/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+package() {
+	cd "$builddir"/build
+	make DESTDIR="$pkgdir" install || return 1
+	rm -rf "$pkgdir"/usr/share/fonts/oiio
+}
+
+doc() {
+	default_doc
+}
```
Comment (jirutka) [inline testing/openimageio/APKBUILD]: `default_doc` is called implicitly when `doc` function is not declared, so you can just omit it.

File: testing/openimageio/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+dev() {
+	default_dev
+}
```
Comment (jirutka) [inline testing/openimageio/APKBUILD]: `default_dev` is called implicitly when `dev` function is not declared, so you can just omit it.

File: testing/openimageio/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+builddir="$srcdir"/oiio-Release-$pkgver
+
+build() {
+	cd "$builddir"
+	mkdir build && cd build
+	cmake .. \
+		-DUSE_OPENSSL=ON \
+		-DUSE_PYTHON=ON \
+		-DUSE_PYTHON3=OFF \
```
Comment (jirutka) [inline testing/openimageio/APKBUILD]: Why have you disabled Python 3?

File: testing/openimageio/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+pkgdesc="Image i/o library supporting a multitude of image formats"
+url="https://sites.google.com/site/openimageio/"
+arch="all"
+license="BSD"
+depends=""
+makedepends="cmake giflib-dev zlib-dev boost-dev jpeg-dev libpng-dev tiff-dev
+	glew-dev python2-dev openexr-dev ffmpeg2.8-dev libraw-dev libwebp-dev freetype-dev
+	qt-dev mesa-dev openssl-dev jasper-dev opencolorio-dev"
+install=""
+subpackages="$pkgname-python2:py2 $pkgname-python3:py3 $pkgname-dev $pkgname-doc $pkgname-utils"
```
Comment (jirutka) [inline testing/openimageio/APKBUILD]: Python subpackages should be named `py2-$pkgname` and `py3-$pkgname`.

Comment (jirutka) [inline testing/openimageio/APKBUILD]: You can also use common split function `_python` for both of these subpackages. Then you can use `$subpkgname` in the split function to get name of the specific subpackage and e.g. `${subpkgname:2:1}` to extract `2` or `3`.

**Comment by Klowner** [comment]
It seems there's some sort of issue with the boost packages? I was having this same build issue on my local machine, but after manually rebuilding boost everything seems to have resolved itself. 😕

**Comment by jirutka** [comment]
I'm on vacation now, with almost no internet connection. I'll finish it at the end of week.

**Comment by Klowner** [comment]
enjoy your vacation! ⛵

**Comment by Klowner** [comment]
Anything else left for this? I think I got everything.

**Pull request closed without merging**